--- origsrc/zlib-1.3.2/win32/Makefile.gcc	2026-02-17 12:47:06.000000000 +0000
+++ src/zlib-1.3.2/win32/Makefile.gcc	2026-09-27 20:02:04.081197300 +0100
@@ -89,7 +89,7 @@ $(IMPLIB): $(SHAREDLIB)
 
 $(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
 	$(CC) -shared -static-libgcc -Wl,--out-implib,$(IMPLIB) $(LDFLAGS) \
-	-o $@ win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
+	-o $@ $^
 	$(STRIP) $@
 
 example.exe: example.o $(STATICLIB)
@@ -109,13 +109,13 @@ minigzip_d.exe: minigzip.o $(IMPLIB)
 	$(STRIP) $@
 
 example.o: test/example.c zlib.h zconf.h
-	$(CC) $(CFLAGS) -I. -c -o $@ test/example.c
+	$(CC) $(CFLAGS) -I. -c -o $@ $<
 
 minigzip.o: test/minigzip.c zlib.h zconf.h
-	$(CC) $(CFLAGS) -I. -c -o $@ test/minigzip.c
+	$(CC) $(CFLAGS) -I. -c -o $@ $<
 
 zlibrc.o: win32/zlib1.rc
-	$(RC) $(RCFLAGS) -o $@ win32/zlib1.rc
+	$(RC) $(RCFLAGS) -o $@ $<
 
 .PHONY: install uninstall clean
 
