--- origsrc/zlib-1.2.8/win32/Makefile.gcc	2012-05-02 22:17:59.000000000 -0500
+++ src/zlib-1.2.8/win32/Makefile.gcc	2013-06-28 16:20:57.206694900 -0500
@@ -94,7 +94,7 @@ $(IMPLIB): $(SHAREDLIB)
 
 $(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
 	$(CC) -shared -Wl,--out-implib,$(IMPLIB) $(LDFLAGS) \
-	-o $@ win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
+	-o $@ $^
 	$(STRIP) $@
 
 example.exe: example.o $(STATICLIB)
@@ -114,13 +114,13 @@ minigzip_d.exe: minigzip.o $(IMPLIB)
 	$(STRIP) $@
 
 example.o: test/example.c zlib.h zconf.h
-	$(CC) $(CFLAGS) -I. -c -o $@ test/example.c
+	$(CC) $(CFLAGS) -I. -c -o $@ $<
 
 minigzip.o: test/minigzip.c zlib.h zconf.h
-	$(CC) $(CFLAGS) -I. -c -o $@ test/minigzip.c
+	$(CC) $(CFLAGS) -I. -c -o $@ $<
 
 zlibrc.o: win32/zlib1.rc
-	$(RC) $(RCFLAGS) -o $@ win32/zlib1.rc
+	$(RC) $(RCFLAGS) -o $@ $<
 
 .PHONY: install uninstall clean
 
